### PR TITLE
feat: Introduce Whispir actions and triggers for Zapier integration

### PR DESCRIPTION
**Pull request by whutomo**
**Summary**
This pull request contains the development work of Zapier application for Whispir use cases which includes the following: 
- Creation of hook triggers when sent message from the Whispir platform is undelivered or when the recipients of the message have replied.
- Actions that allow users to set up a Zapier flow to send messages (e.g. SMS messages, Email messages and messages from templates)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        "type": "git",
+        "url": "https://github.com/GIT_USER_ID/GIT_REPO_ID.git"
+    },
+    "main": "dist/api.js",
+    "types": "dist/api.d.ts",
+    "scripts": {
+        "clean": "rm -Rf node_modules/ *.js",
+        "build": "tsc",
+        "test": "zapier test",
+        "push": "zapier push"
```
Comment (whispir-sharie) [inline package.json]: Hi @whutomo  Can we add the following scripts to support the dev't and testing of our integration

- zapier login
- zapier logs
- zapier versions
- zapier logs --type=http
- zapier logs --type=http --detailed --limit=1
- zapier validate

Comment (whispir-sharie) [reply package.json]: Also might me good to ensure that all is good before the push so can we add a zapier validate & test before executing the zapier push? 

e.g  ` "push": "zapier validate && zapier test &&  zapier push"`

File: triggers/messageReply.ts [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+      true: true,
+      false: false,
+    },
+  },
+  {
+    key: "email",
+    required: true,
+    label: "Email",
+    helpText:
+      "Whispir notify this email address that a callback has failed (only when retries are disabled)",
```
Comment (whispir-sharie) [inline triggers/messageReply.ts]: Can we update to `notifies` and `when`

`"Whispir notifies this email address when a callback has failed (only when retries are disabled)",`

File: triggers/messageUndelivered.ts [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+  } = bundle;
+
+  const payload = {
+    id: uuidv4(),
+    messageId,
+    recipientName: name,
+    recpientMobileNumber: mobile,
+    recipientEmail: email,
+    recipientVoice: voice,
+    recpientResponseContent: content,
```
Comment (whispir-sharie) [inline triggers/messageUndelivered.ts]: @whutomo  Should we have `recpientResponseContent` mapped in undelivered? Or is this only fitting for messageReply received?

**Review by whispir-sharie — CHANGES_REQUESTED** [review]
Great work! I have minor requests and questions 👍

File: triggers/messageReply.ts (hunk 2 above)
Comment (whutomo) [reply triggers/messageReply.ts]: yes, I'll update this. Thanks sharie for pointing it out

File: triggers/messageUndelivered.ts (hunk 3 above)
Comment (whutomo) [reply triggers/messageUndelivered.ts]: Good catch, I think we should not include it here given it's for undelivered message. So, there won't be replies available for Zapier to process. I'll remove it. Thanks Sharie

File: package.json (hunk 1 above)
Comment (whutomo) [reply package.json]: Good call, I have added `zapier validate` before running `zapier push` in the package.json script. Would it be better to add the rest of the commands as part of the readme document to contribute/maintain to whispir-zapier project? it feels quite redundant to me if we were to re-write Zapier commands within the package script so the devs could run `yarn login` to run `zapier login` for an example.

Comment (whispir-sharie) [reply package.json]: Good point from your end as well @whutomo , just wanted to make it easier for developers so they won't have to look outside of repo for testing the work. But definitely we should add it on `readme` page :) 

maybe just the add `login` and `logs` (simple one) is something we can add so there could be a bit of idea already on package.json for a developer to use?

File: triggers/messageUndelivered.ts (hunk 3 above)
Comment (whispir-sharie) [reply triggers/messageUndelivered.ts]: @whutomo  thanks! Before we resolve this comment, is there other payload attribute that we can add in `undelivered`? Like status or time of the status was updated?

**Review by Fatal-Errol — APPROVED** [review]

**Review by whispir-sharie — APPROVED** [review]
Amazing work, Harry 👾

**Pull request merged**